testsuite/Makefile

```diff
@@ -155,7 +155,10 @@ default:
 	@echo "  one TEST=f      launch just this single test"
 	@echo "  one DIR=p       launch the tests located in path p"
 	@echo "  one LIST=f      launch the tests listed in f (one per line)"
-	@echo "  promote DIR=p   promote the reference files for the tests in p"
+	@echo "  promote TEST=f  promote the output for this single test"
+	@echo "  promote DIR=p   promote the output for the tests located in path p"
+	@echo "  promote LIST=f  promote the output for the tests listed in f (one \
+	per line)"
```

Comment:
This is getting a bit wordy with 6 lines, it could possibly be refactored something like:

```
	@echo "  one DIR=p       launch the tests located in path p"
	@echo "  promote DIR=p   promote the output for the tests located in path p"
	@echo "                  (both one and promote accept TEST=f for a single test and
	@echo "                  LIST=f for a list of tests in f (one per line)"
```

Reply:
What about something like this?
```
Available targets:
  all                         launch all tests
  all-foo                     launch all tests beginning with foo
  parallel                    launch all tests using GNU parallel
  parallel-foo                launch all tests beginning with foo using GNU parallel
  one (TEST|DIR|LIST)=x       launch the specified tests
  promote (TEST|DIR|LIST)=x   promote the output for the specified tests
    ... TEST=f                specify this single test
    ... DIR=p                 specify the tests located in path p
    ... LIST=f                specify the tests listed in f (one per line)
  lib                         build library modules
  tools                       build test tools
  clean                       delete generated files
  report                      print the report for the last execution
```
This has the "parallel-foo" line at over 80 characters though, and it only saves one line.  Even the version you propose only saves two lines, and it hides two very useful options inside a parenthetical.  I'm not sure there's a good option that's clearer than just listing all the targets directly.  We could also try grouping them into sections, or adding line breaks to allow for more easily following what's there?

Reply:
The precise text I went with is

```
Available targets:
  all                        launch all tests
  all-foo                    launch all tests beginning with foo
  parallel                   launch all tests using GNU parallel
  parallel-foo               launch all tests beginning with foo using GNU parallel
  one (TEST|DIR|LIST)=x      launch the specified tests ...
  promote (TEST|DIR|LIST)=x  promote the output for the specified tests ...
    ... TEST=f                 ... the single test in the file f
    ... DIR=d                  ... the tests located in the directory d
    ... LIST=f                 ... the tests listed in the file f (one per line)
  lib                        build library modules
  tools                      build test tools
  clean                      delete generated files
  report                     print the report for the last execution

all*, parallel* and list can automatically re-run failed test
directories if MAX_TESTSUITE_DIR_RETRIES permits
(default value = )
```

My only potential concern is that the `parallel-foo` line is >80 columns (by 3); my only proposed fix is changing "launch" to "run" as the verb for what we do with tests, but I'm open to other ideas (e.g., `starting` is 1 character less than `beginning`).